crates/sim2h/src/mono_ref.rs

```diff
@@ -0,0 +1,110 @@
+use crate::*;
+
+/// ref counted - lets us override some debugging
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct MonoRef<T>(std::sync::Arc<T>);
+
+impl std::fmt::Debug for MonoRef<AgentId> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "AgentId({})", self.0)
+    }
+}
+impl std::fmt::Debug for MonoRef<SpaceHash> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "SpaceHash({})", self.0)
+    }
+}
+impl std::fmt::Debug for MonoRef<EntryHash> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "EntryHash({})", self.0)
+    }
+}
+impl std::fmt::Debug for MonoRef<AspectHash> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "AspectHash({})", self.0)
+    }
+}
+impl std::fmt::Debug for MonoRef<Lib3hUri> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Uri({})", self.0)
+    }
+}
+
+/*
+// WTF - why can't we make a generic one too?
+impl<T: std::fmt::Debug> std::fmt::Debug for MonoRef<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self.0)
+    }
+}
+*/
```

Comment:
It's possible todo this but sadly you have to turn a feature flag on 

#![feature(specialization)]

Which would also require you to put a 'default' before fn fmt

Reply:
ah, gottcha - well, it's not strictly necessary at the moment. Hopefully that'll become stable some day : )

Reply:
added/updated the comment to include a note about the feature :+1: